datacube/cfg/api.py

```diff
@@ -10,14 +10,22 @@
 import warnings
 
 from os import PathLike
-from typing import Any, Union   # Union required as typehint | operator doesn't work on string forward references
+from typing import Any, TypeAlias, Union
 
 from .cfg import find_config, parse_text
 from .exceptions import ConfigException
 from .opt import ODCOptionHandler, AliasOptionHandler, IndexDriverOptionHandler
 from .utils import ConfigDict, check_valid_env_name
 
 
+# TypeAliases for more concise type hints
+# (Unions required as typehint | operator doesn't work with string forward-references.
+GeneralisedPath: TypeAlias = str | PathLike | list[str | PathLike]
+GeneralisedCfg: TypeAlias = Union["ODCConfig", GeneralisedPath, None]
+GeneralisedEnv: TypeAlias = Union["ODCEnvironment", str, None]
+GeneralisedRawCfg: TypeAlias = str | ConfigDict | None
```

Comment:
I feel like these TypeAliases should be consistent in whether or not they include `None`

Reply:
I can see that.  It gets a little awkward where they are nested, but there's no real harm in or-ing with None twice.